spec/arguments_parser_spec.rb

```diff
@@ -0,0 +1,41 @@
+require './lib/arguments_parser'
+
+RSpec.describe ArgumentsParser do
+    subject(:arguments_parser) { ArgumentsParser.new(args) }
+
+    let(:args) { [] }
+
+    describe "#parse" do
+        subject(:parse) { arguments_parser.parse }
+        
+        context 'when showing a gem' do
+            let(:args) { ['show', 'rspec'] }
+
+            it "shows a gem details" do  
+                allow(arguments_parser).to receive(:handle_show).with ('rspec')
+
+                arguments_parser.parse
+                arguments_parser.execute
+
+                expect(arguments_parser).to have_received(:handle_show).with ('rspec')
```

Comment:
Very cool that you've thought about using `have_received`.

In unit testing there is a say that one should not test the implementation, unfortunately mocking like the above code does test the implementation.

One obvious problem with this is that one can't easily refactor the code now because if you restructure you class and the functionality is good the test will fail because maybe the name's changed.

Try to refactor your specs to test the actuall results of the functions, if that now works easily then refactor the code.